airbyte-integrations/connectors/source-freshdesk/source_freshdesk/streams.py

```diff
@@ -0,0 +1,254 @@
+        self._state = None
+
+    @property
+    def cursor_field(self) -> str:
+        return "updated_at"
+    
+    @property
+    def state(self) -> MutableMapping[str, Any]:
+        return {"updated_at": self._state} if self._state else {}
+
+    @state.setter
+    def state(self, value: MutableMapping[str, Any]):
+        if "updated_at" in value:
+            self._state = value["updated_at"]
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        if "updated_at" in stream_state:
+            params[self.state_filter] = stream_state["updated_at"]
+        else:
+            params[self.state_filter] = self.start_date
+        return params
+    
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        for record in super().read_records(sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state):
+            yield record
+            if not self.state or self.state["updated_at"] < record[self.cursor_field]:
+                self.state = record
+
+
+class Agents(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "agents"
+
+
+class Companies(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "companies"
+
+
+class Contacts(IncrementalFreshdeskStream):
+    state_filter = "_updated_since"
+    state_checkpoint_interval = 100
+
+    def path(self, **kwargs) -> str:
+        return "contacts"
+
+
+class Groups(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "groups"
+
+
+class Roles(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "roles"
+
+
+class Skills(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "skills"
+
+
+class TimeEntries(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "time_entries"
+
+
+class Surveys(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "surveys"
+
+
+class Tickets(IncrementalFreshdeskStream):
+    ticket_paginate_limit = 300
+    state_checkpoint_interval = 100
+
+    def path(self, **kwargs) -> str:
+        return "tickets"
+    
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        params.update({
```

Comment:
Do you think this is can also be set in the `IncrementalFreshdeskStream` so that we have the same ordering behavior for all incremental streams?

Reply:
Not every API supports these parameters. Most of them only accept the pagination ones.